pkg/compute/serve.go

```diff
@@ -0,0 +1,278 @@
+func NewServeCommand(parent common.Registerer, globals *config.Data, build *BuildCommand, viceroyVersioner update.Versioner) *ServeCommand {
+	var c ServeCommand
+
+	c.build = build
+	c.viceroyVersioner = viceroyVersioner
+
+	c.Globals = globals
+	c.CmdClause = parent.Command("serve", "Build and run a Compute@Edge package locally")
+
+	c.manifest.File.SetOutput(c.Globals.Output)
+	c.manifest.File.Read(manifest.Filename)
+
+	// Build flags
+	c.CmdClause.Flag("name", "Package name").Action(c.name.Set).StringVar(&c.name.Value)
+	c.CmdClause.Flag("language", "Language type").Action(c.lang.Set).StringVar(&c.lang.Value)
+	c.CmdClause.Flag("include-source", "Include source code in built package").Action(c.includeSrc.Set).BoolVar(&c.includeSrc.Value)
+	c.CmdClause.Flag("force", "Skip verification steps and force build").Action(c.force.Set).BoolVar(&c.force.Value)
+
+	return &c
+}
+
+// Exec implements the command interface.
+func (c *ServeCommand) Exec(in io.Reader, out io.Writer) (err error) {
+	// Reset the fields on the BuildCommand based on ServeCommand values.
+	if c.name.WasSet {
+		c.build.PackageName = c.name.Value
+	}
+	if c.lang.WasSet {
+		c.build.Lang = c.lang.Value
+	}
+	if c.includeSrc.WasSet {
+		c.build.IncludeSrc = c.includeSrc.Value
+	}
+	if c.force.WasSet {
+		c.build.Force = c.force.Value
+	}
+
+	err = c.build.Exec(in, out)
+	if err != nil {
+		return err
+	}
+
+	text.Break(out)
+
+	progress := text.NewQuietProgress(out)
+
+	bin, err := getViceroy(progress, out, c.viceroyVersioner)
+	if err != nil {
+		return err
+	}
+
+	err = local(bin, out)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getViceroy returns the path to the installed binary.
+//
+// NOTE: if viceroy is installed then it is updated, otherwise download the
+// latest version and install it in the same directory as the application
+// configuration data.
+func getViceroy(progress text.Progress, out io.Writer, versioner update.Versioner) (string, error) {
+	progress.Step("Checking latest viceroy release...")
+
+	latest, err := versioner.LatestVersion(context.Background())
+	if err != nil {
+		progress.Fail()
+
+		return "", errors.RemediationError{
+			Inner:       fmt.Errorf("error fetching latest version: %w", err),
+			Remediation: errors.NetworkRemediation,
+		}
+	}
+
+	bin := filepath.Join(InstallDir, versioner.Name())
+
+	// gosec flagged this:
+	// G204 (CWE-78): Subprocess launched with variable
+	// Disabling as the variables come from trusted sources.
+	/* #nosec */
+	cmd := exec.Command(bin, "--version")
+
+	stdoutStderr, err := cmd.CombinedOutput()
+
+	if err != nil {
+		// We presume an error executing `viceroy --version` means it isn't installed.
+		//
+		// NOTE: we can't use exec.LookPath("viceroy") because PATH is unreliable
+		// across OS platforms but also we actually install viceroy in the same
+		// location as the application configuration, which means it wouldn't be
+		// found looking up by the PATH env var.
```

Comment:
@phamann The following thoughts all feel like I'm worrying about stuff that's too low-level, hence the implementation I've used doesn't try to concern itself. But that said, there are two things I'm wondering about this section:

1. I don't know if my comment "PATH is unreliable across OS platforms" is _**100%**_ accurate.

2. re: "it wouldn't be found looking up by the PATH env var" should we consider adding the config directory to the PATH (again, I'm not sure how cross platform `os.Setenv()` is**)?

> ** looking at the [`syscall.Setenv`](https://golang.org/src/syscall/env_unix.go?s=1956:1992#L86) implementation it has an OS check for Plan9 but no other OS which might not necessarily mean an issue for Windows.